app/models/project_step.rb

```diff
@@ -42,6 +45,9 @@ class NoChildrenAllowedError < StandardError; end
   SUPER_EARLY_PERIOD = 7.0 # days
   SUPER_LATE_PERIOD = 30.0 # days
 
+  belongs_to :schedule_ancestor,   class_name: 'ProjectStep'
+  has_many   :schedule_decendants, class_name: 'ProjectStep', foreign_key: :schedule_ancestor_id
```

Comment:
Also, I believe the proper naming for these fields would be parent and child/children. ancestor/descendant implies multiple generations, which is not the case here.